schema/test/unit/views/ciip_carbon_tax_calculation_test.sql

```diff
@@ -1,68 +1,89 @@
 set client_encoding = 'utf-8';
 set client_min_messages = warning;
 create extension if not exists pgtap;
-reset client_min_messages;
 
 begin;
-select plan(3);
-
-select test_helper.modify_triggers('disable', '{"form_result": ["_immutable_form_result"]}');
-
-update ggircs_portal.application set reporting_year = 2018 where id = 1;
+select plan(2);
 
 -- View should exist
 select has_view(
     'ggircs_portal', 'ciip_carbon_tax_calculation',
     'ggircs_portal.ciip_carbon_tax_calculation should be a view'
 );
 
+truncate ggircs_portal.application restart identity cascade;
+select test_helper.modify_triggers('disable');
+reset client_min_messages;
+select test_helper.mock_open_window(2019);
+select ggircs_portal.create_application_mutation_chain(1);
+select test_helper.mock_open_window(2020);
+select ggircs_portal.create_application_mutation_chain(1);
+select test_helper.mock_open_window(2021);
+select ggircs_portal.create_application_mutation_chain(1);
+
+-- update all three applications with the same amounts of fuels
 update ggircs_portal.form_result
-  set form_result = '[
-    {
-      "fuelRowId": 11,
-      "quantity": 400,
-      "fuelUnits": "kL",
-      "emissionCategoryRowId": 2
-    }
-  ]'
-  where form_id = 3
-  and application_id = 1
-  and version_number = 1;
+set form_result = json_build_array(
```

Comment:
:+1: nice. I hadn't thought to use json_build_* functions to create the form results